test/e2e/flowaggregator_test.go

```diff
@@ -691,9 +719,32 @@ func getUnit64FieldFromRecord(t *testing.T, record string, field string) uint64
 	return 0
 }
 
-// TODO: Add a test that checks the functionality of Pod-To-External flow.
-func checkFlowType(t *testing.T, record string, flowType uint8) {
-	assert.Containsf(t, record, fmt.Sprintf("flowType: %d", flowType), "Record does not have correct flowType")
+func getExternalServerIPv4(t *testing.T, server string) string {
+	ips, _ := net.LookupIP(server)
+	var ipv4 string
+	for _, ip := range ips {
+		if ip.To4() != nil {
+			ipv4 = ip.String()
+		}
+	}
+	return ipv4
+}
+
+func getCollectorOutput(t *testing.T, srcIP string, dstIP string, timeStart time.Time) string {
+	err := wait.Poll(250*time.Millisecond, collectorCheckTimeout, func() (bool, error) {
+		rc, collectorOutput, _, err := provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl logs --since=%v ipfix-collector -n antrea-test", time.Since(timeStart).String()))
+		if err != nil || rc != 0 {
+			return false, err
+		}
+		return strings.Contains(collectorOutput, srcIP) && strings.Contains(collectorOutput, dstIP), nil
+	})
+	require.NoErrorf(t, err, "IPFIX collector did not receive the expected records and timed out with error: %v", err)
+
+	rc, collectorOutput, _, err := provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl logs --since=%v ipfix-collector -n antrea-test", time.Since(timeStart).String()))
+	if err != nil || rc != 0 {
+		t.Errorf("Error when getting logs %v, rc: %v", err, rc)
+	}
```

Comment:
Can't we reuse the L493-L515 code with a better timeout value? Is there any reason we have to wait for 12 seconds?

Reply:
Updated the util function to reuse the L493-L515 code. And differentiated the cases 1) where we need to check all records are received, specific to iperf traffic, 2) or we only want to check one single record. 
Previously I reused the code from `checkRecordsForDenyFlows`. Just confirmed with yiou, we can unify the timeout to 6s.